src/record/json-path.js

```diff
@@ -51,19 +82,22 @@ JsonPath.prototype.setValue = function (node, value) {
  * @returns {void}
  */
 JsonPath.prototype._tokenize = function () {
-  const parts = this._path.split(SPLIT_REG_EXP)
+
+  // makes json path array items a single 'part' value of parts below
+  // 'arrayProp[#]' members transform to 'arrayProp=#' now instead of 'arrayProp.#' previously
+  // see setValue fnc above for special handling of array item parsing vs numeric obj member name
+  // e.g. 'object.1' parsing. this allows for support of parsing and differentiating object
+  // member names that are also numeric values
+  let str = this._path.replace(/\s/g, '')
```

Comment:
this is super expensive, doing three inline regexs and not even compiling them all upfront.